### PR TITLE
Fix stretching issue in comment section on shifting detail page

### LINKED ISSUE #6637: Comment Section in the shifting detail page is stretching the page
**Describe the bug**

When we post multiple comment on a shifting detail page, it keep on stretching the page to the bottom

**To Reproduce**

1. Go to any existing shifting detail page and post multiple comments ( https://care.coronasafe.in/shifting/e785d651-9fde-4279-ac33-e9282296dc10 )
2. See error

**Expected behavior**

It shouldn't be stretching the page a lot and the entire page should be effectively utilizing the white space

**Screenshots**

image

### PR DESCRIPTION
**Pull request by Thanush19**
I have realigned the UI of the shift page and also added a "Read More" and "Read Less" option for comments with more than 30 words.

Closes #6637

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Kindly link the issue in the PR using closing keywords and give a more targetted title for the PR (eg: "Fixes comment section in Shifting Details")

The PR title would come up in the changelog.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by Thanush19** [comment]
hey @rithviknishad , i haved renamed the PR and Issue  i have resolved is Comment Section in the shifting detail page is stretching the page #6637 ( https://github.com/coronasafe/care_fe/issues/6637)

**Comment by rithviknishad** [comment]
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

1. Use closing keywords
2. Should be in the main comment of the PR

For example, see: https://github.com/coronasafe/care_fe/pull/7358

**Comment by rithviknishad** [comment]
Doing so will cause the PR to be linked to the issue and once the PR is merged, issue gets closed automatically.

**Comment by bodhish** [comment]
I think we should publish a small section in the readme about commit messages;

**Comment by Thanush19** [comment]
is the PR title is okay now , if any changes need to be done , intimate me @rithviknishad ...

File: src/Components/Shifting/CommentsSection.tsx [hunk 1]
```diff
@@ -9,10 +9,68 @@ import { IComment } from "../Resource/models";
 import PaginatedList from "../../CAREUI/misc/PaginatedList";
 import request from "../../Utils/request/request";
 
-interface CommentSectionProps {
+interface CommentProps {
   id: string;
+  comment: string;
+  created_by_object: any;
```
Comment (rithviknishad) [inline src/Components/Shifting/CommentsSection.tsx:15]: Can we have proper types here? We have a pre-defined type: "PerformedByUserModel" which can be used here.

File: src/Components/Shifting/CommentsSection.tsx [hunk 2]
```diff
@@ -9,10 +9,68 @@ import { IComment } from "../Resource/models";
...
+            <button onClick={toggleExpanded} className="ml-2 text-green-600">
+              ..... Read more
+            </button>
+          )}
+          {expanded && remainingComment && (
+            <button onClick={toggleExpanded} className="ml-2 text-green-600">
+              Read less
+            </button>
+          )}
+        </p>
```
Comment (rithviknishad) [inline src/Components/Shifting/CommentsSection.tsx:53]: Would it be possible to re-use the already implemented `Readmore` component here? https://github.com/coronasafe/care_fe/blob/develop/src/Components/Common/components/Readmore.tsx

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Shifting/CommentsSection.tsx (hunk 1 above)
Comment (Thanush19) [reply src/Components/Shifting/CommentsSection.tsx:15]: The 'PerformedByUserModel' interface does not have the required type that I need for the 'commentSection', so I can't use it here. Therefore, I defined a new interface.
export type UserBareMinimum = {
  id: number;
  username: string;
  first_name: string;
  last_name: string;
  email: string;
  user_type: UserRole;
  last_login: string | undefined;
};


interface CommentProps {
  id: string;
  comment: string;
  created_by_object: any;
  modified_date: string;
}

Comment (rithviknishad) [reply src/Components/Shifting/CommentsSection.tsx:15]: It was already type-safe before you did the modifications. You've removed the types... 

image

File: src/Components/Shifting/CommentsSection.tsx [hunk 3]
```diff
@@ -9,10 +9,68 @@ import { IComment } from "../Resource/models";
...
+  return (
+    <div
+      key={id}
+      className="mt-4 flex  flex-col rounded-lg border border-gray-500 bg-white p-4 text-gray-800"
+    >
+      <div className="flex  w-full ">
+        <p className="text-justify">
+          {expanded ? comment : truncatedComment}
+          {!expanded && remainingComment && (
+            <button onClick={toggleExpanded} className="ml-2 text-green-600">
```
Comment (rithviknishad) [inline src/Components/Shifting/CommentsSection.tsx:44]: Use `primary` color instead of `green`

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
Closing as duplicate of #7402

**Pull request closed without merging**